Source/DafnyCore/Verifier/Translator.cs

```diff
@@ -6686,10 +6687,11 @@ AND  Apply(f,h0,bxs) == Apply(f,h0,bxs)
           return predef.ArrayLength;
         } else if (f.EnclosingClass is ValuetypeDecl { Name: "real" } && f.Name == "Floor") {
           return predef.RealFloor;
-        } else if (f is SpecialField && !(f is DatatypeDestructor)) {
+        } else if (f is SpecialField && !(f is DatatypeDestructor || f.EnclosingClass is IndDatatypeDecl or ClassLikeDecl)) {
```

Comment:
You're checking `IndDatatypeDecl` instead of `DatatypeDecl`. The latter also includes codatatype declarations. Did you intend to make this distinction? I don't quite understand the test, but perhaps you even want `TopLevelDeclWithMembers` instead of `IndDatatypeDecl or ClassLikeDecl`, which would then also include abstract types and newtypes. I suggest adding test cases for these, too.

Reply:
Interesting remark. Testing it with co-datatype did not yield any error, but opaque types and newtypes did yield errors.
However, map itself being a ValueTypeDecl, which is a TopLevelDeclWithMembers, I had to make an exclusion check, but it ended up working.